panther_utils/panther_utils/arguments.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+# Copyright 2024 Husarion sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Text, Tuple
+
+import yaml
+from launch.action import Action
+from launch.actions import DeclareLaunchArgument
+from launch.launch_context import LaunchContext
+from launch.substitutions import EnvironmentVariable, LaunchConfiguration
+from launch.utilities import normalize_to_list_of_substitutions, perform_substitutions
+
+# Define valid configurations for each robot model
+VALID_CONFIGURATIONS = {
+    "panther": {"wheel_type": ["WH01", "WH02", "WH04"]},
+    "lynx": {"wheel_type": ["WH05"]},
+}
+
+
+class DeclareRobotArgs(Action):
+    """Retrieves and validate the robot configuration from the YAML data."""
+
+    def __init__(self, robot_configuration_path: LaunchConfiguration, **kwargs) -> None:
+        """Create a DeclareRobotArgs action."""
+        super().__init__(**kwargs)
+        self.__path = normalize_to_list_of_substitutions(robot_configuration_path)
+
+    @property
+    def path(self) -> Text:
+        """Getter for self.__path."""
+        return self.__path
+
+    def execute(self, context: LaunchContext):
+        """Execute the action."""
+        path = perform_substitutions(context, self.path)
+        yaml_data = self.load_yaml_file(path)
+        yaml_data = self.normalize_robot_configuration(yaml_data)
+        namespace, robot_config = self.extract_first_robot_configuration(yaml_data)
+        list_of_args = self.create_launch_arguments(namespace, robot_config)
+        for arg in list_of_args:
+            arg.execute(context)
+
+    def load_yaml_file(self, path: str) -> dict:
```

Comment:
```suggestion
    def _load_yaml_file(self, path: str) -> dict:
```
Private methods should begin with underscore

Reply:
not actual